packages/vite/src/node/cli.ts

```diff
@@ -50,6 +51,19 @@ function cleanOptions(options: GlobalCLIOptions) {
   return ret
 }
 
+function listenRestart(server: ViteDevServer) {
+  process.stdin.resume()
+  process.stdin.setEncoding('utf8')
+  process.stdin.on('data', async (data) => {
+    const str = data.toString().trim().toLowerCase()
+    // if the keys entered match the restartable value, then restart!
+    // we can make the restart word configurable if necessary
+    if (str === 'rs') {
```

Comment:
Let's use [clikey](https://github.com/zewish/clikey) for this, instead.

Then we can do the following:
- `r` to restart server
- `f` to restart server with `--force` flag
- `o` to open browser

Reply:
We also want to print the possible shortkeys at first start up

Reply:
> Let's use clikey for this, instead.

I am not sure about if we want to introduce a new dep for this, despite the existing code works well.

Reply:
this line https://github.com/zewish/clikey/blob/master/index.js#L9 will [cause `ctrl + c`  no longer a `SIGINT`](https://nodejs.org/api/tty.html#tty_readstream_setrawmode_mode)

maybe we should  get rid of it

Reply:
@antfu Sounds good 👍 

I think we want `.setRawMode(true)` so that input isn't printed to console.

We can do this to handle `ctrl+c` 👇 

```ts
// ctrl + c
if (str == '\x03') {
  console.log('')
  return process.kill(process.pid, 'SIGINT')
}
```

Reply:
@aleclarson  so are we gonna using this dep ?   it constraints one character only which not extensible I think, we may make the keyword configurable

Reply:
@antfu said we shouldn't, and I agree. Just need the `.setRawMode(true)` part.

>  it constraints one character only which not extensible I think, we may make the keyword configurable

I doubt it, as I don't see the benefit in customizing the key bindings. Not sure how the others feel about it.

Reply:
Was `setRawMode` not needed at the end?